plugins/repository-s3/src/test/java/org/elasticsearch/repositories/s3/S3BlobStoreContainerTests.java

```diff
@@ -19,21 +19,62 @@
 
 package org.elasticsearch.repositories.s3;
 
+import org.apache.logging.log4j.Level;
+import org.apache.logging.log4j.Logger;
 import org.elasticsearch.common.blobstore.BlobStore;
+import org.elasticsearch.common.logging.Loggers;
 import org.elasticsearch.common.settings.Settings;
 import org.elasticsearch.common.unit.ByteSizeUnit;
 import org.elasticsearch.common.unit.ByteSizeValue;
+import org.elasticsearch.mocksocket.MockServerSocket;
 import org.elasticsearch.repositories.ESBlobStoreContainerTestCase;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
 
 import java.io.IOException;
+import java.net.InetAddress;
+import java.net.ServerSocket;
 import java.util.Locale;
 
 public class S3BlobStoreContainerTests extends ESBlobStoreContainerTestCase {
+
+    private static final Logger logger = Loggers.getLogger(S3BlobStoreContainerTests.class);
+
+    private static ServerSocket mockS3ServerSocket;
+
+    private static Thread mockS3AcceptorThread;
+
+    // Opens a MockSocket to simulate connections to S3 checking that SocketPermissions are set up correctly.
+    // See MockAmazonS3.simulateS3SocketConnection.
+    @BeforeClass
+    public static void openMockSocket() throws IOException {
+        mockS3ServerSocket = new MockServerSocket(0, 50, InetAddress.getByName("127.0.0.1"));
+        mockS3AcceptorThread = new Thread(() -> {
+            while (!mockS3ServerSocket.isClosed()) {
+                try {
+                    // Accept connections from MockAmazonS3.
+                    mockS3ServerSocket.accept();
+                } catch (IOException e) {
+                    logger.log(Level.ERROR, e);
```

Comment:
I think that this will generate an error in the logs most of the time when this test ends. As we will call `close` and I think `accept` throws an exception when the socket is closed. @jasontedor - is this alright?